File: iBurn/src/main/java/com/gaiagps/iburn/database/PlayaItem.java
package com.gaiagps.iburn.database;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.PrimaryKey;

import com.mapbox.mapboxsdk.geometry.LatLng;

import java.io.Serializable;

/**
 * Created by dbro on 6/8/17.
 */

public class PlayaItem implements Serializable {
    public static final String ID = "_id";
    public static final String NAME = "name";
    public static final String DESC = "desc";
    public static final String URL = "url";
    public static final String CONTACT = "contact";
    public static final String PLAYA_ADDR = "p_addr";
    public static final String PLAYA_ADDR_UNOFFICIAL = "p_addr_unof";
    public static final String PLAYA_ID = "p_id";
    public static final String LATITUDE = "lat";
    public static final String LONGITUDE = "lon";
    public static final String LATITUDE_UNOFFICIAL = "lat_unof";
    public static final String LONGITUDE_UNOFFICIAL = "lon_unof";
    public static final String FAVORITE = "fav";


    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = ID)
    public int id;

    @ColumnInfo(name = NAME)
    public String name;

    @ColumnInfo(name = DESC)
    public String description;

    @ColumnInfo(name = URL)
    public String url;

    @ColumnInfo(name = CONTACT)
    public String contact;

    @ColumnInfo(name = PLAYA_ADDR)
    public String playaAddress;

    @ColumnInfo(name = PLAYA_ADDR_UNOFFICIAL)
    public String playaAddressUnofficial;

    @ColumnInfo(name = PLAYA_ID)
    public String playaId;

    @ColumnInfo(name = LATITUDE)
    public float latitude;

    @ColumnInfo(name = LONGITUDE)
    public float longitude;

    @ColumnInfo(name = LATITUDE_UNOFFICIAL)
    public float latitudeUnofficial;

    @ColumnInfo(name = LONGITUDE_UNOFFICIAL)
    public float longitudeUnofficial;

    @ColumnInfo(name = FAVORITE)
    public boolean isFavorite;

    public boolean hasLocation() {
        return latitude != 0 && longitude != 0;
    }

    public boolean hasUnofficialLocation() {
        return latitudeUnofficial != 0 && longitudeUnofficial != 0;
    }

    public LatLng getLatLng() {
        return new LatLng(latitude, longitude);
    }

    public LatLng getUnofficialLatLng() {
        return new LatLng(latitudeUnofficial, longitudeUnofficial);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PlayaItem playaItem = (PlayaItem) o;

        if (id != playaItem.id) return false;
        return playaId != null ? playaId.equals(playaItem.playaId) : playaItem.playaId == null;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (playaId != null ? playaId.hashCode() : 0);
        return result;
    }
}

File: iBurn/src/main/java/com/gaiagps/iburn/database/PlayaItem.kt
package com.gaiagps.iburn.database

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize
import org.maplibre.android.geometry.LatLng

/**
 * Created by dbro on 6/8/17.
 */
@Parcelize
open class PlayaItem : Parcelable {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = ID)
    var id: Int = 0

    @JvmField
    @ColumnInfo(name = NAME)
    var name: String? = null

    @JvmField
    @ColumnInfo(name = DESC)
    var description: String? = null

    @ColumnInfo(name = URL)
    var url: String? = null

    @ColumnInfo(name = CONTACT)
    var contact: String? = null

    @JvmField
    @ColumnInfo(name = PLAYA_ADDR)
    var playaAddress: String? = null

    @JvmField
    @ColumnInfo(name = PLAYA_ADDR_UNOFFICIAL)
    var playaAddressUnofficial: String? = null

    @JvmField
    @ColumnInfo(name = PLAYA_ID)
    var playaId: String? = null

    @JvmField
    @ColumnInfo(name = LATITUDE)
    var latitude: Float = 0f

    @JvmField
    @ColumnInfo(name = LONGITUDE)
    var longitude: Float = 0f

    @JvmField
    @ColumnInfo(name = LATITUDE_UNOFFICIAL)
    var latitudeUnofficial: Float = 0f

    @JvmField
    @ColumnInfo(name = LONGITUDE_UNOFFICIAL)
    var longitudeUnofficial: Float = 0f

    fun hasLocation(): Boolean {
        return latitude != 0f && longitude != 0f
    }

    fun hasUnofficialLocation(): Boolean {
        return latitudeUnofficial != 0f && longitudeUnofficial != 0f
    }

    val latLng: LatLng
        get() = LatLng(latitude.toDouble(), longitude.toDouble())

    val unofficialLatLng: LatLng
        get() = LatLng(latitudeUnofficial.toDouble(), longitudeUnofficial.toDouble())

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val playaItem = o as PlayaItem

        if (id != playaItem.id) return false
        return if (playaId != null) (playaId == playaItem.playaId) else playaItem.playaId == null
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + (if (playaId != null) playaId.hashCode() else 0)
        return result
    }

    companion object {
        const val ID: String = "_id"
        const val NAME: String = "name"
        const val DESC: String = "desc"
        const val URL: String = "url"
        const val CONTACT: String = "contact"
        const val PLAYA_ADDR: String = "p_addr"
        const val PLAYA_ADDR_UNOFFICIAL: String = "p_addr_unof"
        const val PLAYA_ID: String = "p_id"
        const val LATITUDE: String = "lat"
        const val LONGITUDE: String = "lon"
        const val LATITUDE_UNOFFICIAL: String = "lat_unof"
        const val LONGITUDE_UNOFFICIAL: String = "lon_unof"
    }
}
